src/catchup/ApplyBucketsWork.cpp

```diff
@@ -25,7 +25,8 @@ namespace stellar
 ApplyBucketsWork::ApplyBucketsWork(
     Application& app,
     std::map<std::string, std::shared_ptr<Bucket>> const& buckets,
-    HistoryArchiveState const& applyState, uint32_t maxProtocolVersion)
+    HistoryArchiveState const& applyState, uint32_t maxProtocolVersion,
```

Comment:
Great point. I did a check-up on the diff, and tightened the protocol versions in two places so that new-style merge changes are independent of maxProtocolVersion parameters:
* `Bucket::merge` now uses `protocolVersion` to update metrics, and perform some proto 12 sanity checks
* `BucketList::restartMerges` uses snap's protocol version to create new FutureBucket, so new-style merge restart is independent of the `maxProtocolVersion` parameter.

I also changed FutureBucket to take `protocolVersion` in its constructor, since in both usages in the code we know the exact version to use (specifically, in `BucketList::prepare` and `BucketList::restartMerges`). The remaining usages of `maxProtocolVersion` are still there, as introduced by protocol 11.

Reply:
Looks like this is fixed, thanks.